lib/web_ui/lib/src/engine/view_embedder/dom_manager.dart

```diff
@@ -47,38 +137,58 @@ class DomManager {
   /// The tag name for the scene host.
   static const String sceneHostTagName = 'flt-scene-host';
 
+  /// The tag name for the text editing host.
+  static const String textEditingHostTagName = 'flt-text-editing-host';
+
   /// The tag name for the semantics host.
   static const String semanticsHostTagName = 'flt-semantics-host';
 
   /// The tag name for the accessibility announcements host.
   static const String announcementsHostTagName = 'flt-announcement-host';
 
-  final FlutterViewEmbedder _embedder;
-
   /// The root DOM element for the entire Flutter View.
   ///
   /// This is where input events are captured, such as pointer events.
   ///
   /// If semantics is enabled, this element also contains the semantics DOM tree,
   /// which captures semantics input events.
-  DomElement get rootElement => _embedder.flutterViewElementDEPRECATED;
+  DomElement rootElement;
 
   /// Hosts all platform view elements.
-  DomElement get platformViewsHost => _embedder.glassPaneElementDEPRECATED;
+  DomElement platformViewsHost;
 
   /// Hosts all rendering elements and canvases.
-  DomShadowRoot get renderingHost => _embedder.glassPaneShadowDEPRECATED;
+  DomShadowRoot renderingHost;
+
+  /// Hosts the <flt-scene> element.
+  ///
+  /// This element is created and inserted in the HTML DOM once. It is never
+  /// removed or moved. However the <flt-scene> inside of it may be replaced.
+  DomElement sceneHost;
 
   /// Hosts all text editing elements.
-  DomElement get textEditingHost => _embedder.textEditingHostNodeDEPRECATED;
+  DomElement textEditingHost;
 
   /// Hosts the semantics tree.
   ///
   /// This element is in front of the [renderingHost] and [platformViewsHost].
   /// Otherwise, the phone will disable focusing by touch, only by tabbing
   /// around the UI.
-  DomElement get semanticsHost => _embedder.semanticsHostElementDEPRECATED;
+  DomElement semanticsHost;
 
   /// This is where accessibility announcements are inserted.
-  DomElement get announcementsHost => _embedder.announcementsHostDEPRECATED;
+  DomElement announcementsHost;
+}
+
+DomShadowRoot _attachShadowRoot(DomElement element) {
+  if (getJsProperty<Object?>(element, 'attachShadow') == null) {
```

Comment:
I think we can remove this check now. I believe @ditman removed the non-shadow DOM mode a long time ago.

Reply:
It was me who removed it, and I kept this check so that we show a somewhat meaningful error to the user instead of an obscure null check error. I'm happy to remove it if we think there's no value in having this error message.

Reply:
There's probably some value, but also, this leaks into production code and increases code size, so there's also value in removing it.

Reply:
Ok, let me put it behind an assert.